### PR TITLE
Added as_hdf() function

### PR DESCRIPTION
**Pull request by Jonathan-J-Deng**
Hi!

I added a function to convert the TDMS file to an HDF5 file format.

I think the formats map to each other very nicely and I wanted to do this but saw there was no function for it.

I'm not a super good coder or anything and haven't contributed to open source stuff before so let me know if I should change anything, or if I'm doing this wrong.

Thanks!

### CODE REVIEW

File: nptdms/tdms.py [hunk 1]
```diff
@@ -225,6 +225,46 @@ def as_dataframe(self, time_index=False, absolute_time=False):
...
+
+        # First right the properties at the root level
+        for property_name, property_value in self.object().properties.items():
+            h5file['/'].attrs[property_name] = property_value
+
+        # Now iterate through groups and channels,
+        # writing the properties and data
+        for group_name in self.groups():
+
+            group = self.object(group_name)
```
Comment (adamreeve) [inline nptdms/tdms.py]: One thing to be careful of is that there may be a group in the file without an associated group object, eg. it's possible to have a channel at `/'group_name'/'channel_name'` but no group object for `/'group_name'`. Please check for a `KeyError` when getting the group object. And above when you get the root object you should also check for a `KeyError`. You can see an example of something similar in https://github.com/adamreeve/npTDMS/blob/master/nptdms/tdmsinfo.py.

File: nptdms/tdms.py [hunk 2]
```diff
@@ -225,6 +225,46 @@ def as_dataframe(self, time_index=False, absolute_time=False):
...
+
+        # Channels in TDMS are then mapped to the second level of the HDF5
+        # hierarchy, under the appropriate groups.
+
+        # Properties in TDMS are mapped to attributes in HDF5.
+        # These all exist under the appropriate, channel group etc.
+
+        h5file = h5py.File(filepath, 'w')
+
+        # First right the properties at the root level
```
Comment (adamreeve) [inline nptdms/tdms.py]: Minor nitpick, this should say "write", not "right"

**Comment by adamreeve** [comment]
Thanks for contributing this, I agree the formats are a good match and this will probably be useful for other people too. I just have a couple of comments above that need fixing then I'll merge it.

File: nptdms/tdms.py [hunk 3]
```diff
@@ -225,6 +225,54 @@ def as_dataframe(self, time_index=False, absolute_time=False):
...
+        for group_name in self.groups():
+            try:
+                group = self.object(group_name)
+
+                # Write the group's properties
+                for prop_name, prop_value in group.properties.items():
+                    h5file['/'+group_name].attrs[prop_name] = prop_value
+
+                # Write properties and data for each channel
+                for channel in self.group_channels(group_name):
```
Comment (adamreeve) [inline nptdms/tdms.py]: This section should be outside the try block. There may be channels in the group so `self.group_channels(group_name)` works but `self.object(group_name)` could fail.

Comment (Jonathan-J-Deng) [inline nptdms/tdms.py]: Okay, I move it out. Just wondering, why isn't a TDMS object always created for the groups/root?

Comment (adamreeve) [inline nptdms/tdms.py]: Good question, I think LabView usually creates TDMS files with group objects but I have come across some that were missing. We could create an empty group object automatically in npTDMS if there isn't one already, which might be a good idea.

**Pull request merged**